akka-cluster/src/test/scala/akka/cluster/RemoteFeaturesWithClusterSpec.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster
+
+import akka.actor.Actor
+import akka.actor.ExtendedActorSystem
+import akka.remote.RemoteFeaturesSpec
+import com.typesafe.config.ConfigFactory
+
+object RemoteFeaturesWithClusterSpec {
+  class TestActor extends Actor {
+    def receive: Receive = { case _ => }
+  }
+}
+
+class RemoteFeaturesWithClusterSpec
+    extends RemoteFeaturesSpec(
+      ConfigFactory.parseString("""akka.actor.provider = "cluster"""").withFallback(RemoteFeaturesSpec.disabled)) {
+
+  "Remoting with Cluster" must {
+
+    "have the correct settings" in {
+      system.settings.HasCluster shouldBe true
+      provider.hasClusterOrUseUnsafe shouldBe true
+      provider.transport.system.settings.HasCluster shouldBe true
+      provider.remoteSettings.UseUnsafeRemoteFeaturesWithoutCluster shouldBe false
+      system.settings.ProviderClass shouldEqual classOf[ClusterActorRefProvider].getName
+    }
+
+    "create a ClusterRemoteWatcher" in {
+      provider.remoteWatcher.isDefined shouldBe true
+    }
+
+    "use ClusterDeployer" in {
+      provider.deployer.getClass == classOf[ClusterDeployer] shouldBe true
+    }
+
+    "have expected `actorOf` behavior" in {
+      val conf = s"""
+        ${RemoteFeaturesSpec.common(false)}
+        akka.actor.provider = "cluster"
+        akka.remote.artery.canonical.port = 0"""
+
+      val selfAddress = system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
```

Comment:
`Cluster(system).selfAddress` is nicer here

Reply:
Right, I must have had that from a remote test and forgot.